lib/fromZodError.ts

```diff
@@ -1,23 +1,22 @@
-import {
-  ISSUE_SEPARATOR,
-  MAX_ISSUES_IN_MESSAGE,
-  PREFIX,
-  PREFIX_SEPARATOR,
-  UNION_SEPARATOR,
-} from './config.ts';
-import { getMessageFromZodIssue } from './fromZodIssue.ts';
-import { prefixMessage } from './prefixMessage.ts';
-import { ValidationError } from './ValidationError.ts';
 import { fromError } from './fromError.ts';
 import { isZodErrorLike } from './isZodErrorLike.ts';
+import {
+  createMessageBuilder,
+  type CreateMessageBuilderProps,
+  type MessageBuilder,
+} from './MessageBuilder.ts';
+import { isNonEmptyArray } from './utils/NonEmptyArray.ts';
+import { ValidationError } from './ValidationError.ts';
 import type * as zod from 'zod';
-import type { FromZodIssueOptions } from './fromZodIssue.ts';
 
 export type ZodError = zod.ZodError;
 
-export type FromZodErrorOptions = FromZodIssueOptions & {
-  maxIssuesInMessage?: number;
-};
+export type FromZodErrorOptions =
```

Comment:
[question] Do we need to make them mutually exclusive? i.e either pass `messageBuilder` or any other option from the `CreateMessageBuilderProps` ?

Reply:
Hm, don't we? I actually think this makes for better DX.